### PR TITLE
[#1510] Import bulk forms

### LINKED ISSUE #1510: Bulk import forms
### Thema / Theme

Admin

### Omschrijving / Description

Related to #1482 (bulk export forms)

There should be the functionality to import forms exported in bulk

### Added value / Toegevoegde waarde

_No response_

### Aanvullende opmerkingen / Additional context

_No response_

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #1510

### CODE REVIEW

File: src/openforms/forms/admin/views.py [hunk 1]
```diff
@@ -54,3 +65,64 @@ def get(self, request, *args, **kwargs):
...
+    def _bulk_import_forms(self, import_file):
+        imports_dir = Path(settings.PRIVATE_MEDIA_ROOT, "imports")
+        if not imports_dir.exists():
+            os.mkdir(imports_dir)
+        imports_file = Path(imports_dir, f"import_forms_{uuid4()}.zip")
+        with open(imports_file, "wb") as f_private:
+            for chunk in import_file.chunks():
+                f_private.write(chunk)
+
+        process_forms_import.delay(str(imports_file), self.request.user.id)
```
Comment (sergei-maertens) [inline src/openforms/forms/admin/views.py:128]: I'd completely delegate this to the file storage:

```py
from privates.storages import private_media_storage

name = f"imports/import_forms_{uuid4()}.zip"
name = private_media_storage.save(name, import_file)

process_forms_import.delay(name, self.request.user.id)
```

File: src/openforms/forms/admin/tasks.py [hunk 2]
```diff
@@ -70,6 +75,29 @@ def process_forms_export(forms_uuids: list, user_id: int) -> None:
         )
 
 
+@app.task
+def process_forms_import(import_file: str, user_id: int) -> None:
+    user = User.objects.get(id=user_id)
+    failed_files = []
+    # This deletes the temp dir once the context manager is exited
+    with tempfile.TemporaryDirectory(dir=Path(import_file).parent) as temp_dir:
+        with zipfile.ZipFile(import_file, "r") as zip_file:
```
Comment (sergei-maertens) [inline src/openforms/forms/admin/tasks.py:84]: I'd leverage the storage class here too, following the changes in the view:

```py
from privates.storages import private_media_storage

def process_forms_import(filename: str, user_id: int) -> None:
    # This deletes the temp dir once the context manager is exited
    with tempfile.TemporaryDirectory(dir=Path(import_file).parent) as temp_dir:
        with zipfile.ZipFile(private_media_storage.open(filename), "r") as zip_file:
            ...
```

File: src/openforms/forms/admin/tasks.py [hunk 3]
```diff
@@ -70,6 +75,29 @@ def process_forms_export(forms_uuids: list, user_id: int) -> None:
...
+                        zip_file.extract(member=zipped_form_file, path=temp_dir)
+                    )
+                except ValidationError as exc:
+                    filename = Path(zipped_form_file.filename).name
+                    failed_files.append((filename, exc.detail))
+                    logger.error("Could not import form %s", filename)
+                    continue
+
+    logevent.bulk_forms_imported(user=user, failed_files=failed_files)
+    os.remove(import_file)
```
Comment (sergei-maertens) [inline src/openforms/forms/admin/tasks.py:98]: should use the storage class too: `private_media_storage.delete(filename)`

File: src/openforms/forms/tests/admin/test_tasks.py [hunk 4]
```diff
@@ -60,3 +66,65 @@ def test_zip_file_contains_data(self):
...
+
+        process_forms_export(
+            forms_uuids=[form1.uuid, form2.uuid],
+            user_id=user.id,
+        )
+
+        cls.form_export = FormsExport.objects.get()
+        cls.user = user
+
+    def _copy_file_to_imports_tempdir(self):
```
Comment (sergei-maertens) [inline src/openforms/forms/tests/admin/test_tasks.py:86]: better to use the private_media_storage here too

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/openforms/forms/admin/tasks.py (hunk 2 above)
Comment (sergei-maertens) [reply src/openforms/forms/admin/tasks.py:84]: you can pass a filename or any file-like object to `ZipFile`, which is what's happening in my suggestion. We open it explicitly using the storage, and the storage is responsible for finding it on file system (or possibly in the future in Amazon S3 or other object storages)

It's basically using the abstraction that uploaded files may not be on the same server in the future, and this code pattern can handle both situations.

Comment (SilviaAmAm) [reply src/openforms/forms/admin/tasks.py]: Ohh i see, i shall update my update

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**